src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -36,6 +36,31 @@
        (map (comp :raw-balance val))
        (reduce money/add)))
 
+(defn extract-exponent
+  [s]
+  (if-let [index (string/index-of s "e")]
+    (subs s (+ index 2))
+    nil))
+
+(defn calc-max-crypto-decimals
+  [value]
+  (let [str-representation (str value)
+        decimal-part       (second (clojure.string/split str-representation #"\."))
+        exponent           (extract-exponent str-representation)
+        zeroes-count       (count (take-while #(= \0 %) decimal-part))
+        max-decimals       (or exponent zeroes-count)]
+    (let [first-non-zero-digit (first (filter #(not (= \0 %)) decimal-part))]
+      (if (= \1 first-non-zero-digit)
+        (inc max-decimals)
+        max-decimals))))
+
+(defn prettify-crypto
+  [{:keys [market-values-per-currency]} token-units]
+  (let [price          (get-in market-values-per-currency [:usd :price])
```

Comment:
We have to use the user's preferred/selected currency for calculation. `USD` is the default one.

Can we get the price from the function it is called `calc-token-value` (from `market-values`  de-structuring)? it takes the user's preferred/selected currency.

Reply:
@smohamedjavid For calculating the max decimals places we always use the value of one cent of a USD. You can see full description here: https://github.com/status-im/status-mobile/issues/18244

Reply:
Oh! Apologies. I overlooked. Thanks for adding the quote, @OmarBasem.

Can you help add a small note in that function? It will help a lot.

Reply:
Actually @OmarBasem - it might be worth adding a comment in this file with the reference to that GitHub issue 🤔

Reply:
Or we could make a readme doc in the root of wallet if we think any more pieces about the wallet need some docs